### PR TITLE
Remove Ubuntu 18.04 from CI, add Ubuntu 24.04

### PR DESCRIPTION
**Pull request by tammojan**

### CODE REVIEW

File: docker/ubuntu2404_gcc.docker [hunk 1]
```diff
@@ -42,10 +42,9 @@ RUN cmake .. \
     -DBUILD_TESTING=ON \
     -DUSE_OPENMP=OFF \
     -DUSE_HDF5=ON \
-    -DBUILD_PYTHON=ON \
-    -DBUILD_PYTHON3=ON \
     -DCMAKE_PREFIX_PATH=${CMAKE_PREFIX_PATH} \
     -DDATA_DIR=/usr/local/share/casacore/data \
+    -DBUILD_FFTPACK_DEPRECATED=ON \
```
Comment (gmloose) [inline docker/ubuntu2404_gcc.docker:47]: Not sure why you'd still want to use FFTPack and then something which is deprecated. Isn't FFTW3 the best choice nowadays? But it's fine with me.

**Review by gmloose — COMMENTED** [review]
LGTM. See my comment about FFTPack.

**Review by gmloose — APPROVED** [review]
LGTM. See my remark about FFTPack.

File: docker/ubuntu2404_gcc.docker (hunk 1 above)
Comment (tammojan) [reply docker/ubuntu2404_gcc.docker:47]: I simply copied `ubuntu_2004` to `ubuntu_2204`.

Comment (gmloose) [reply docker/ubuntu2404_gcc.docker:47]: OK, fine. But maybe something to consider for later.

**Pull request merged**